src/utils/api.js

```diff
@@ -6,12 +6,25 @@ export const getArticleViaId = (article_id) => {
   });
 };
 
-export const getAllArticles = (topic) => {
+export const getAllArticles = (topic, sortByQuery, orderByQuery) => {
   // console.log(topic);
+  let url = `/articles${topic ? `?topic=${topic}` : ""}`;
+
+  if (sortByQuery && orderByQuery) {
+    url += `${topic ? "&" : "?"}sort_by=${sortByQuery}&order=${orderByQuery}`;
+  } else if (sortByQuery) {
+    url += `${topic ? "&" : "?"}sort_by=${sortByQuery}`;
+  } else if (orderByQuery) {
+    url += `${topic ? "&" : "?"}order=${orderByQuery}`;
+  }
```

Comment:
This is **not necessary** axios allows you to add query params via the options object
https://masteringjs.io/tutorials/axios/get-query-params

It takes out the need to check whether you need a '?' or '&', it will ignore it if the param is undefined, it will do a lot of hard work for you. Please refactor this accordingly